packages/theme/src/cli/services/profile.ts

```diff
@@ -0,0 +1,67 @@
+import {openURL} from '@shopify/cli-kit/node/system'
+import {ensureAuthenticatedStorefront} from '@shopify/cli-kit/node/session'
+import {joinPath} from '@shopify/cli-kit/node/path'
+import {writeFile} from 'fs/promises'
+import {tmpdir} from 'os'
+
+export async function profile(password: string | undefined, storeDomain: string, urlPath: string, asJson: boolean) {
+  // Fetch the profiling from the Store
+  const url = new URL(`https://${storeDomain}${urlPath}`)
+  const storefrontToken = await ensureAuthenticatedStorefront([], password)
+  const response = await fetch(url, {
+    headers: {
+      Authorization: `Bearer ${storefrontToken}`,
+      Accept: 'application/vnd.speedscope+json',
+    },
+  })
+  const contentType = response.headers.get('content-type')
+  if (response.status !== 200 && contentType !== 'application/json') {
+    throw new Error(`Bad response: ${response.status}, content-type: ${contentType}`)
+  }
+
+  const profileJson = await response.text()
+
+  if (asJson) {
+    // Print the JSON
+    process.stdout.write(profileJson)
+  } else {
+    await openProfile(profileJson)
+  }
+}
+
+async function resolveSpeedscope() {
+  if (import.meta.resolve) {
+    return import.meta.resolve('speedscope/dist/release/index.html')
+  } else {
+    try {
+      const speedscopePath = require.resolve('speedscope/package.json')
+      const speedscopeDir = speedscopePath.replace('/package.json', '')
+      return `file://${speedscopeDir}/dist/release/index.html`
+    } catch (error) {
+      throw new Error("Can't find Speedscope package")
+    }
+  }
```

Comment:
Seems to work both in tests and with `dev shopify`. Is there some packaging magic that could make this not work?